ksql-engine/src/test/java/io/confluent/ksql/planner/plan/KsqlStructuredDataOutputNodeTest.java

```diff
@@ -189,4 +191,47 @@ private SchemaKStream buildStream() {
         new HashMap<>(), new MockSchemaRegistryClient());
   }
 
+  @Test
+  public void shouldCreateSinkWithCorrectCleanupPolicy() {
+    KafkaTopicClient topicClientForNonWindowTable = EasyMock.createNiceMock(KafkaTopicClient.class);
+    final Map<String, Object> props = new HashMap<>();
+    props.put(KsqlConfig.SINK_NUMBER_OF_PARTITIONS_PROPERTY, 4);
+    props.put(KsqlConfig.SINK_NUMBER_OF_REPLICAS_PROPERTY, (short)3);
+
+    StructuredDataSourceNode tableSourceNode = new StructuredDataSourceNode(
+        new PlanNodeId("0"),
+        new KsqlTable("sqlExpression", "datasource",
+                      schema,
+                      schema.field("key"),
+                      schema.field("timestamp"),
+                      new KsqlTopic("input", "input",
+                                     new KsqlJsonTopicSerDe()),
+                      "TableStateStore",
+                      false),
+        schema);
+
+    outputNode = new KsqlStructuredDataOutputNode(new PlanNodeId("0"),
+                                                  tableSourceNode,
+                                                  schema,
+                                                  schema.field("timestamp"),
+                                                  schema.field("key"),
+                                                  new KsqlTopic("output", "output", new KsqlJsonTopicSerDe()),
+                                                  "output",
+                                                  props,
+                                                  Optional.empty());
+
+    StreamsBuilder streamsBuilder = new StreamsBuilder();
+    SchemaKTable schemaKTable = (SchemaKTable) outputNode.buildStream(streamsBuilder,
```

Comment:
We should first have a check `assertThat(scheamKstream, instanceof(schemaKTable))` and then do the cast.